src/components/product/product-extractor.service.ts

```diff
@@ -26,11 +26,12 @@ export class ProductExtractor {
     const sheet = pnp.planning;
 
     const stepColumns = findStepColumns(sheet, availableSteps);
+    const progressStepColumns = findStepColumns(pnp.progress);
```

Comment:
I think availableSteps should be passed here too
```suggestion
    const progressStepColumns = findStepColumns(pnp.progress, availableSteps);
```